File: src/main/java/fi/eonwe/wikilinks/fatpages/PagePointer.java
package fi.eonwe.wikilinks.fatpages;

import javax.annotation.Nullable;

/**
*/
public class PagePointer {

    private static int counter = 0;

    @Nullable
    public WikiPage page;
    public final int id = counter++;

    public PagePointer(@Nullable WikiPage page) {
        this.page = page;
    }

    @Override
    public int hashCode() {
        return id;
    }

    @Override
    public boolean equals(Object obj) {
        return (obj instanceof PagePointer && ((PagePointer) obj).id == id);
    }
}

File: src/main/kotlin/fi/eonwe/wikilinks/fatpages/PagePointer.kt
package fi.eonwe.wikilinks.fatpages

/**
 */
class PagePointer(var page: WikiPage?) {
    val id: Int = counter++

    override fun hashCode(): Int {
        return id
    }

    override fun equals(other: Any?): Boolean {
        return (other is PagePointer && other.id == id)
    }

    companion object {
        private var counter = 0
    }
}